src/Symfony/Component/ObjectMapper/Tests/ObjectMapperTest.php

```diff
@@ -291,4 +295,15 @@ public function testMultipleTargetMapProperty()
         $this->assertEquals($c->foo, 'donotmap');
         $this->assertEquals($c->doesNotExistInTargetB, 'foo');
     }
+
+    public function testTransformCollection()
+    {
+        $u = new TransformCollectionA();
+        $u->foo = [new TransformCollectionC('a'), new TransformCollectionC('b')];
+        $mapper = new ObjectMapper();
+
+        $transformed = $mapper->map($u, TransformCollectionB::class);
+
+        $this->assertEquals($transformed->foo, [new TransformCollectionD('a'), new TransformCollectionD('b')]);
```

Comment:
The first parameter is the expected value.
```suggestion
        $this->assertEquals([new TransformCollectionD('a'), new TransformCollectionD('b')], $transformed->foo);
```